crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -121,9 +126,10 @@ pub fn token_tree_to_macro_items(
 /// Parses the token tree (result of macro expansion) as a sequence of items
 pub fn token_tree_to_ast_item_list(tt: &tt::Subtree) -> TreeArc<ast::SourceFile> {
     let buffer = tt::buffer::TokenBuffer::new(&[tt.clone().into()]);
-    let token_source = SubtreeTokenSource::new(&buffer);
-    let mut tree_sink = TtTreeSink::new(token_source.querier());
-    ra_parser::parse(&token_source, &mut tree_sink);
+    let mut token_source = SubtreeTokenSource::new(&buffer);
+    let querier = token_source.querier();
+    let mut tree_sink = TtTreeSink::new(querier.as_ref());
+    ra_parser::parse(&mut token_source, &mut tree_sink);
```

Comment:
I wonder if these 6 lines could be extracted into a helper function?

Reply:
Fixed.